endpoints/decorators/metrics.go

```diff
@@ -27,21 +31,29 @@ func assignMetricsFunctions(m *metrics.Metrics, method int) *metricsFunctions {
 		metrics.RecordDuration = m.RecordPutDuration
 		metrics.RecordBadRequest = m.RecordPutBadRequest
 		metrics.RecordError = m.RecordPutError
+		metrics.RecordKeyProvided = m.RecordPutKeyProvided
 	case GetMethod:
 		metrics.RecordTotal = m.RecordGetTotal
 		metrics.RecordDuration = m.RecordGetDuration
 		metrics.RecordBadRequest = m.RecordGetBadRequest
 		metrics.RecordError = m.RecordGetError
+		metrics.RecordKeyProvided = func() {} // Doesn't get called from Get endpoint
 	}
 	return metrics
 }
 
 type writerWithStatus struct {
-	delegate   http.ResponseWriter
-	statusCode int
+	delegate       http.ResponseWriter
+	statusCode     int
+	wasKeyProvided bool
 }
 
 func (w *writerWithStatus) WriteHeader(statusCode int) {
+	if statusCode == CacheUpdateCode {
```

Comment:
This seems hacky.

Reply:
Lol. It's meant to store the info we need to know whether or not to call `RecordKeyProvided()` in line 84 below. Want me to remove and try another way?

Reply:
May be better to add a method like `FlagKeyProvided()` onto this struct to allow recording the status, would resolve confusion on what header `CacheUpdateCode` is supposed to represent.

Reply:
We get an error. Moved it to the `Write(bytes []byte)` method instead